tests/programs/custom/debug_test_trigger/debug_test_trigger.c

```diff
@@ -1611,27 +1655,23 @@ int test_register_access(void) {
   __asm__ volatile (R"(li    s1,     0x0
                        csrw  tinfo,  s1
                        csrr  s0,     tinfo
-                       li    s1,     0x8064
+                       li    s1,     0x01008064
                        bne   s0,     s1, 1f
                        li    s1,     0x0   #SUCCESS
                        sw    s1,     g_register_access_status, s2
                      1:nop
                        )" ::: "s0", "s1", "s2");
   if (g_register_access_status != SUCCESS) return FAIL;
 
-  // TCONTROL - Write 1s
-  g_register_access_status = FAIL;
-  __asm__ volatile (R"(li    s1,     0xFFFFFFFF
-                       csrw  tcontrol,  s1
-                       csrr  s0,     tcontrol
-                       bnez  s0,     1f
-                       li    s1,     0x0   #SUCCESS
-                       sw    s1,     g_register_access_status, s2
-                     1:nop
-                       )" ::: "s0", "s1", "s2", "memory");
-  if (g_register_access_status != SUCCESS) return FAIL;
-
   // Context Registers - Access Checks (in machine mode)
+  g_illegal_insn_status = 0;
+  __asm__ volatile ("csrwi tcontrol, 0x0");
+  if (!g_illegal_insn_status) return FAIL;
+
+  g_illegal_insn_status = 0;
+  __asm__ volatile ("csrwi tdata3, 0x0");
+  if (!g_illegal_insn_status) return FAIL;
```

Comment:
I don't know the trigger registers, but the self-checking looks good, because it sets itself up for failure and it can only avoid failure if the assembly executes according to expectations.

Reply:
^ tdata3 was removed a while back